dts: arm: st: add missing or correct #address-cells property in exti node

etienne-lms opened the pull request:
Add missing #address-cells property in exti interrupt controller node for l5xx and wbaxx SoCs.
Prevents build warning trace messages like:
```
.../zephyr/build/zephyr/zephyr.dts:130.39-186.5: Warning (interrupt_provider): /soc/interrupt-controller@4000f400: Missing #address-cells in interrupt provider

mathieuchopstm commented:
Where is this error message even coming from? I can't find this message in `dtc`'s source code
<img width="962" height="340" alt="image" src="https://github.com/user-attachments/assets/a7a2c155-acf6-42ae-b9ee-953cad263dc6" />

etienne-lms commented:
That depends on DTC version. The test was introduced in DTC v1.6.1 (https://github.com/dgibson/dtc/commit/81e0919a3e211844245a3c19be28610f930c13c0)
then removed in DTC v1.7.0 (https://github.com/dgibson/dtc/commit/d8d1a9a77863a8c7031ae82a1d461aa78eb72a7b). It was last restored only for when a interrupt provider provides a `interrupt-map` (https://github.com/dgibson/dtc/commit/0a3a9d3449c88aabe88685ea99811bf6c02a570d and https://github.com/dgibson/dtc/commit/267efc7d469477927baa5c0e540d91cb51ca25bf). None of the currently supported STM32 SoCs do use `interrupt-map`.

In the end, should we be consistent among SoCs with already set `#address-cells = <1>`, or move to `#address-cells = <0>`, or remove them? Sticking to latest released DT schemas, the [property is still expected](https://github.com/devicetree-org/dt-schema/blob/v2025.08/dtschema/schemas/interrupt-controller.yaml#L18) despite some DTC tools may ignore its absence.

etienne-lms commented:
Well, looking again in the bindings, STM32 EXTI compatible drivers are defined through interrupt controllers bindings so it would not really make sense to remove their interrupt controller. I'd rather update the `#address-cells` property to value 0 to match DTC v1.6.1 as well as former and later releases.

etienne-lms committed:
dts: arm: st: l5: add missing #address-cells in exti node

Add missing #address-cells property in exti interrupt controller node.
This change prevents build warning messages when using DTC v1.6.1. With
former or later DTC versions, missing #address-cells property is ignored
but it remains requires as per DT schemas, e.g. [1]

Link: https://github.com/devicetree-org/dt-schema/blob/v2025.08/dtschema/schemas/interrupt-controller.yaml#L18 [1]
Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
dts: arm: st: wba: add missing #address-cells in exti node

Add missing #address-cells property in exti interrupt controller node.
This change prevents build warning messages when using DTC v1.6.1. With
former or later DTC versions, missing #address-cells property is ignored
but it remains requires as per DT schemas, e.g. [1]

Link: https://github.com/devicetree-org/dt-schema/blob/v2025.08/dtschema/schemas/interrupt-controller.yaml#L18 [1]
Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
dts: arm: st: correct exti #address-cells value

Correct #address-cells property in exti interrupt controller nodes
for STM32 SoCs that defined it to 1 whereas value 0 is more applicable
as that interrupt does not expect sub-node nor interrupt mapping.

No functional changes as the value is ignored. This change rather targets
STM32 SoCs DTSI files consistency.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>
